server/src/test/java/org/elasticsearch/search/query/ScriptScoreQueryTests.java

```diff
@@ -131,6 +131,14 @@ public void testExplainDefaultNoScore() throws IOException {
         assertThat(explanation.getValue(), equalTo(2.0f));
     }
 
+    public void testScriptScoreErrorOnNegativeScore() {
```

Comment:
For me this set of unit tests is quite straightforward + easy to understand. I think yml tests are slower to debug and are most helpful for testing the end-to-end integration. We recently added a set of testing guidelines that I think are really helpful: https://github.com/elastic/elasticsearch/blob/master/TESTING.asciidoc#what-kind-of-tests-should-i-write

Reply:
@jtibshirani Thanks a lot, I missed these guidelines, will study them